src/model/document.ts

```diff
@@ -335,7 +335,8 @@ export class Document implements Feature, Queryable {
       }
       if (feature.kinds.has('import') && options.imported) {
         const imprt = feature as Import;
-        if (options.externalPackages || !Package.isExternal(imprt.url)) {
+        if (options.externalPackages ||
```

Comment:
you could also pull `imprt.document && !Package.isExternal(imprt.document.url)` into a variable `isExternalDocument`